src/worker.cc

```diff
@@ -958,6 +958,8 @@ void from_json(rapidjson::Document& doc, Options::Action action, Api& api) {
 
   // Get the matrix_loctions option and set if sources or targets size is one
   // (option is only supported with one to many or many to one matrix requests)
+  // TODO(nils): Why is that? IMO this makes a lot of sense for a many:many call
+  // of timedistancematrix as well no?
```

Comment:
@dnesbitt61 shouldn't we allow to set the `matrix_locations` also for many:many? I think the same logic applies there as well, faster results if someone only wants a subset of destinations for each origin.